api/v1alpha1/meshgateway.proto

```diff
@@ -0,0 +1,119 @@
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+import "api/v1alpha1/common.proto";
+import "gogoproto/gogo.proto";
+import "google/api/field_behavior.proto";
+import "k8s.io/api/core/v1/generated.proto";
+
+// $schema: istio-operator.api.v1alpha1.MeshGatewaySpec
+// $title: Istio Mesh Gateway Spec
+// $description: Istio Mesh Gateway descriptor
+
+package istio_operator.v2.api.v1alpha1;
+
+option go_package = "github.com/banzaicloud/istio-operator/v2/api/v1alpha1";
+
+// MeshGateway defines an Istio ingress or egress gateway
+//
+// <!-- crd generation tags
+// +cue-gen:MeshGateway:groupName:servicemesh.cisco.com
+// +cue-gen:MeshGateway:version:v1alpha1
+// +cue-gen:MeshGateway:storageVersion
+// +cue-gen:MeshGateway:annotations:helm.sh/resource-policy=keep
+// +cue-gen:MeshGateway:subresource:status
+// +cue-gen:MeshGateway:scope:Namespaced
+// +cue-gen:MeshGateway:resource:shortNames=mgw,meshgw
+// +cue-gen:MeshGateway:printerColumn:name=Age,type=date,JSONPath=.metadata.creationTimestamp,description="CreationTimestamp is a timestamp
+// representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations.
+// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+// Populated by the system. Read-only. Null for lists. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata"
+// +cue-gen:MeshGateway:preserveUnknownFields:false
+// -->
+//
+// <!-- go code generation tags
+// +genclient
+// +k8s:deepcopy-gen=true
+// -->
+message MeshGatewaySpec {
+    // Contains the intended version for the Mesh Gateway.
+    // +kubebuilder:validation:Pattern=^1.
+    string version = 1 [(gogoproto.jsontag) = "version", (google.api.field_behavior) = REQUIRED];
+    // General Mesh gateway configuration
+    MeshGatewayConfiguration MeshGwConfiguration = 2 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = ",inline"];
+    // Service ports to open for the gateway
+    // +kubebuilder:validation:MinItems=0
+    repeated ServicePort ports = 3 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "ports"];
+    // Type of gateway, either ingress or egress
+    // +kubebuilder:validation:Enum=ingress;egress
+    GatewayType type = 4  [(gogoproto.jsontag) = "type"];
+    // Istio CR to which this gateway belongs to
+    NamespacedName istioControlPlane = 5 [(gogoproto.jsontag) = "istioControlPlane,omitempty"];
+}
+
+message MeshGatewayConfiguration {
+    // Standard Kubernetes resource configuration
+    BaseK8sResourceConfigurationWithHPAWithoutImage BaseK8sResourceConfigWithHPAWithoutImage = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = ",inline"];
+    // Labels to attach to the mesh gateway
+	map<string, string> Labels = 2 [(gogoproto.jsontag) = "labels,omitempty"];
```

Comment:
These labels were always misleading to the end users where to set these, because labels are usually set in every resource outside of the spec and this field is at `spec.labels`. It's fine like this in this PR, I'd suggest to either create a ticket for this or let's just discuss how to solve this better.

Reply:
I'd leave it as is, until we figure out something better. But we can return to the question when doing the reconciliation.

Reply:
Sure, I'm not holding this PR up, but please let's get back to this because I know for sure that this was misleading multiple times for multiple people (once for me as well :) ).